### DISCUSSION TITLE
Toggling buffers with mini-animate loses transparency for several seconds

### DISCUSSION THREAD

**Discussion by alexlitinsky** [nvim-mini/mini.nvim#222] [General]
Here's a video to show what I'm describing: 

https://user-images.githubusercontent.com/86020820/217321116-fb8a13f0-5db1-4298-bae5-97839c3b53fd.mov

Is there any way to prevent the black background from showing up for several seconds? When I disable the plugin, I don't have this issue.

**Comment by echasnovski** [comment]
~~Unfortunately, I can't see a video. It says "No video with supported format and MIME type found". Probably, because I am on Firefox. Could you try mp4?~~ Nevermind, I can see it now. It looks like my next point still stands.

But even before doing this, do you have window animation turned on? By default, 'mini.animate' animates window opening and window closing. Does disabling them help? You can do by incorporating these config options:

```lua
require('mini.animate').setup({
  open = { enable = false },
  close = { enable = false },
})
```

They by default use `winblend` option of floating windows, so they should look like gradually fading and be barely visible. Look at [demo](https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-animate.md#demo) towards the end.

**Reply by alexlitinsky** [reply]
Yep, this worked. As I'm using LazyVim, I just put that statement in the lazy file.

**Comment by sghng** [comment]
@echasnovski Thanks for the clarification. I also encountered this problem. Is it possible to make the current animation compatible with transparent background? I'm not sure if this feature request makes sense, since `winblend` is designed for a "fading" effect, which may not apply to transparent background...

**Reply by echasnovski** [reply]
It is possible to animate using window effects with [`gen_winconfig`](https://github.com/echasnovski/mini.nvim/blob/301ee9f3afa2224b833de7c182574ba6c61e8c6e/doc/mini-animate.txt#L800) and adjust `winblend` option to be constant. Here is [an example](https://github.com/echasnovski/mini.nvim/blob/301ee9f3afa2224b833de7c182574ba6c61e8c6e/doc/mini-animate.txt#L415-L441).

Another approach might be to set `MiniAnimateNormalFloat` highlight groups to not have transparent background. Not sure if that will help, though (can't remember and it might depend on how transparent background is set up).

But my personal suggestion would be to disable open/close animations altogether if you use transparent background.
